### ISSUE TITLE
Select an HTML element wich have a class with a  blank space

### ISSUE THREAD

**Issue by AnaelMobilia** [jhy/jsoup#521]
Hello,

Simple HTML page :
```html
<div class="class">class without space</div>
<div class="class ">class with space</div>
```

The div "class with space" can't be selected without regexp
My tests case from [try.jsoup.org](http://try.jsoup.org) :
 - ```div[class=class]```
=>
```<div class="class">
class without space```
 - ```div[class=class ]```
=>
```<div class="class">
class without space```
 - ```div[class="class"]```
=>
```<div class="class">
class without space```
 - ```div[class="class "]```
=>
```<div class="class">
class without space```
 - ```div[class="class\ "]```
=> Ø
 - ```div[class~=class ]```
=>
```<div class="class">
class with space```

The [HTML DTD] (http://www.w3.org/TR/html4/struct/global.html#adef-class) say "Multiple class names must be separated by white space characters." => a simple class should'nt have a space... ?

Regards,

**Comment by krystiangor** [comment]
Classes are separated by spaces. So both of these:

    <div class="class">class without space</div>
    <div class="class ">class with space</div>

have exactly the same class. That's also why jsoup trims every value.

    this.value = value.trim().toLowerCase();

 That's why all your examples work.

**Comment by AnaelMobilia** [comment]
@krystiangor read again my message : if i use a non regexp selector, only the div "class without space" is returned, not both.

**Comment by krystiangor** [comment]
Sorry. I debugged this behaviour and it seems to be directly caused by method `matches` in `AttributeWithValue`

        public boolean matches(Element root, Element element) {
            return element.hasAttr(key) && value.equalsIgnoreCase(element.attr(key));
        }

It resolves to `"class".equalsIgnoreCase("class ")` and returns false. Maybe we should trim element attribute value before comparing it. Someone more experienced should comment and analyze if it can do any harm.

**Comment by offa** [comment]
**Side note:** With the selector `div.class` you'll get both - just in case you need a quick fix.

**Comment by offa** [comment]
@AnaelMobilia / @krystiangor i've pushed a fix to offa@2f7079a0771fdbee94230e0297f3e6af66916716 

Please checkout that branch and try again.

**Comment by jhy** [comment]
Thanks for the review and detail on this. I have merged the change. This isn't really about classes per-se, as the attribute name/value selector isn't class specific (which is why the div.class mechanism does work, as that is).
